File: src/main/java/com/zuminX/service/Information.java
package com.zuminX.service;

import com.intellij.AbstractBundle;
import com.intellij.openapi.components.ServiceManager;
import org.jetbrains.annotations.PropertyKey;

/**
 * 信息类
 */
public class Information extends AbstractBundle {

  public static final String BUNDLE = "messages.Information";

  public Information() {
    super(BUNDLE);
  }

  /**
   * 获取信息
   *
   * @param key    该消息对应的键
   * @param params 参数
   * @return 信息
   */
  public static String message(@PropertyKey(resourceBundle = BUNDLE) String key, Object... params) {
    return getInstance().getMessage(key, params);
  }

  /**
   * 获取实例对象
   *
   * @return 信息类对象
   */
  private static Information getInstance() {
    return ServiceManager.getService(Information.class);
  }

}

File: src/main/java/com/zuminX/service/Information.kt
package com.zuminX.service

import com.intellij.AbstractBundle
import com.zuminX.utils.getService
import org.jetbrains.annotations.PropertyKey

/**
 * 信息类
 */
class Information : AbstractBundle(BUNDLE) {
  companion object {
    /**
     * 获取信息
     *
     * @param key    该消息对应的键
     * @param params 参数
     * @return 信息
     */
    @JvmStatic
    fun message(@PropertyKey(resourceBundle = BUNDLE) key: String, vararg params: Any): String {
      return Information::class.java.getService().getMessage(key, params)
    }
  }
}

const val BUNDLE = "messages.Information"